framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -907,6 +907,15 @@ public T findOneBy(SearchCriteria<T> sc) {
         return findOneIncludingRemovedBy(sc);
     }
 
+    @Override
+    @DB()
+    public T findOneBy(SearchCriteria<T> sc, final Filter filter) {
```

Comment:
Since we are fetching results including removed entries, this should be renamed as `findOneIncludingRemovedBy`.

Reply:
Or use search in #L915.

Reply:
@vishesh92 we are fetching only non-removed entries here, and the method in line 915 does not limit to one result, as I need. The other method I could use is `findOneBy(final SearchCriteria<T> sc)`, however, I cannot pass an already created filter. That's why I created this method, as I needed to order the results by the `created` column, and limit to only one result.

Reply:
Got it. I got a little confused because of the method names.